lib/jekyll/assets/cached.rb

```diff
@@ -15,8 +15,13 @@ class Cached < Sprockets::CachedEnvironment
       def initialize(env)
         @parent = env
         @jekyll = env.jekyll
+        @resolve_cache = {}
         super env
       end
+
+      def resolve(*args)
+        @resolve_cache[args] ||= super
```

Comment:
You know, last night, I thought this might have been part of the problem.  I'm glad you confirmed it and thanks!